src/mewbot/io/rss.py

```diff
@@ -0,0 +1,514 @@
+
+    site_url: str  # The site according to us
+    entry: feedparser.util.FeedParserDict  # raw entry for later processing
+
+    startup: bool  # Was this feed read as part of first read for any given site?
+
+
+@dataclasses.dataclass
+class RSSOutputEvent(OutputEvent):
+
+    # pylint: disable=too-many-instance-attributes
+    # Want to fully represent the core RSS standard
+
+    title: str  # The title of the item.
+    link: str  # The URL of the item.
+    description: str  # The item synopsis.
+    author: str  # Email address of the author of the item
+    category: str  # Includes the item in one or more categories.
+    comments: str  # URL of a page for comments relating to the item
+    enclosure: str  # URL of a page for comments relating to the item.
+    guid: str  # Describes a media object that is attached to the item.
+    pub_date: str  # Indicates when the item was published.
+    source: str  # The RSS channel that the item came from.
+
+
+class RSSIO(IOConfig):
+
+    _input: Optional[RSSInput] = None
+    _output: None
+
+    _sites: List[str]  # A list of sites to poll for RSS update events
+    _polling_every: int  # Total re-poll time for the sites in the _sites list
+
+    def __init__(
+        self,
+        *args: Optional[Any],
+        polling_every: Union[str, bytes, SupportsInt],
+        **kwargs: Optional[Any],
+    ) -> None:
+        self._logger = logging.getLogger(__name__ + "DesktopNotificationIO")
+
+        # Not entirely sure why, but empty properties in the yaml errors
+        self._logger.info("DesktopNotificationIO received args - %s", args)
+        self._logger.info("DesktopNotificationIO received kwargs - %s", kwargs)
+
+        self._polling_every = int(polling_every)
+
+    @property
+    def sites(self) -> List[str]:
+        return self._sites
+
+    @sites.setter
+    def sites(self, sites: List[str]) -> None:
+        """
+        Presented as a property because updating it requires updating the runner as well.
+        """
+        self._sites = sites
+        if not self._input:
+            return
+        self._input.sites = sites
+
+    def get_inputs(self) -> Sequence[Input]:
+        if not self._input:
+            self._input = RSSInput(self._sites, self._polling_every)
+
+        return [self._input]
+
+    def get_outputs(self) -> Sequence[Output]:
+        return []
+
+
+class RSSInput(Input):
+    """
+    Polling RSS Input source -
+    """
+
+    _logger: logging.Logger
+
+    _sites: List[str]  # A list of sites to poll for RSS update events
+    _polling_every: int  # How often to poll all the given sites (in seconds)
+    sites_started: Set[str]  # sites which have undergone startup
+    sent_entries: Set[str]  # entries which have been put on the wire
+
+    def __init__(
+        self, sites: List[str], polling_every: int, startup_queue_depth: int = 5
+    ) -> None:
+        super(Input, self).__init__()  # pylint: disable=bad-super-call
```

Comment:
```suggestion
        super().__init__()
```
newer python is smarter for single-inheritance; the pylint warning is poorly worded in explaining what it wants you to do

Reply:
ohh - yes. I had totally misunderstood. Corrected.